src/dependency-manager/src/manager.ts

```diff
@@ -580,5 +580,51 @@ export default abstract class DependencyManager {
     return res as ComponentConfig;
   }
 
-  abstract getArchitectContext(graph: DependencyGraph): Promise<any | undefined>;
+  protected generateEnvironmentIngresses(graph: DependencyGraph, components_map: Dictionary<ComponentConfig>): Dictionary<Dictionary<InterfaceSpec>> {
+    const all_external_interfaces: Dictionary<InterfaceSpec> = {};
+    for (const component_config of Object.values(components_map)) {
+      const component_interfaces = component_config.getInterfaces();
+      for (const [component_interface_name, interface_data] of Object.entries(component_interfaces)) {
+        if (!interface_data.host?.startsWith('${{')) {
+          all_external_interfaces[component_interface_name] = interface_data;
+        }
+      }
+    }
+
+    const ingresses: Dictionary<Dictionary<InterfaceSpec>> = {};
+    for (const [component_name, component_config] of Object.entries(components_map)) {
+      const component_interfaces = component_config.getInterfaces();
+      for (const component_interface_name of Object.keys(component_interfaces)) {
+        const tagless_component_name = component_config.getName();
+
+        let to_node = graph.getNodeByRef(`${component_name}-interfaces`);
+        let current_interface_name = component_interface_name;
+        let dependent_nodes = graph.getDependentNodes(to_node);
+        while (dependent_nodes.length) {
+          const dependent_node = dependent_nodes.pop();
+          const edge = graph.edges.find(edge => edge.from === dependent_node.ref && edge.to === to_node.ref);
+          if (edge instanceof IngressEdge && all_external_interfaces[current_interface_name]) {
```

Comment:
Can we use node.is_external instead of refing the components_map? 

If not it sounds more like a problem in the graph generation if we don't have the information required.

Reply:
Sorry, this was just bad naming